typescript-view-swapping/main.ts

```diff
@@ -0,0 +1,42 @@
+interface View extends HTMLDivElement {
+  dataset: {
+    view: string;
+  };
+}
+
+const $tabContainer = document.querySelector('.tab-container');
+const tabs = document.querySelectorAll('.tab');
+const views = document.querySelectorAll('.view');
+
+if (!$tabContainer) throw new Error('$tabContainer query failed');
+if (!tabs) throw new Error('tabs query failed');
+if (!views) throw new Error('views query failed');
+
+$tabContainer.addEventListener('click', handleClick);
+
+function handleClick(event: Event): void {
+  const $eventTarget = event.target as HTMLDivElement;
+
+  if ($eventTarget.matches('.tab')) {
+    tabs.forEach(($tab) => {
+      if ($tab.isSameNode($eventTarget)) {
```

Comment:
`isSameNode` isn't required anymore. This method existed before JavaScript had strict equality, but now `===` can accomplish the same thing. so you can just do `if ($tab === $eventTarget)`